wgengine/netstack/netstack.go

```diff
@@ -913,13 +921,19 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 		}
 	}
 
-	if ns.ForwardTCPIn != nil {
-		c := createConn()
-		if c == nil {
-			return
+	if ns.GetTCPHandlerForFlow != nil {
+		handler, ok := ns.GetTCPHandlerForFlow(clientRemoteAddrPort, dstAddrPort)
+		if ok {
+			if handler == nil {
+				r.Complete(true)
+				return
+			}
+			c := createConn()
+			if c == nil {
+				return
+			}
+			handler(c)
```

Comment:
I think you're missing a `return` after this?

Reply:
Yep! fixed